src/utils/eth_transaction.cairo

```diff
@@ -307,26 +193,43 @@ namespace EthTransaction {
         pedersen_ptr: HashBuiltin*,
         bitwise_ptr: BitwiseBuiltin*,
         range_check_ptr,
-    }(address: felt, account_nonce: felt, tx_data_len: felt, tx_data: felt*) {
+    }(
+        address: felt,
+        account_nonce: felt,
+        r: Uint256,
+        s: Uint256,
+        v: felt,
+        tx_data_len: felt,
+        tx_data: felt*,
+    ) {
         alloc_locals;
-        let (
-            nonce,
-            gas_price,
-            gas_limit,
-            destination,
-            amount,
-            payload_len,
-            payload,
-            msg_hash,
-            v,
-            r,
-            s,
-        ) = decode(tx_data_len, tx_data);
+        let (nonce, _gas_price, _gas_limit, _, _, chain_id, _, _) = decode(tx_data_len, tx_data);
         assert nonce = account_nonce;
-        let (local keccak_ptr: felt*) = alloc();
-        local keccak_ptr_start: felt* = keccak_ptr;
+        assert chain_id = Constants.CHAIN_ID;
+
+        let (local words: felt*) = alloc();
+        let (keccak_ptr: felt*) = alloc();
+        let keccak_ptr_start = keccak_ptr;
         with keccak_ptr {
-            verify_eth_signature_uint256(msg_hash=msg_hash, r=r, s=s, v=v, eth_address=address);
+            // From keccak/cairo_keccak_bigend doc:
+            // > To use this function, split the input into words of 64 bits (little endian).
+            // > Same as keccak, but outputs the hash in big endian representation.
+            // > Note that the input is still treated as little endian.
+            bytes_to_bytes8_little_endian(words, tx_data_len, tx_data);
+            let (msg_hash) = cairo_keccak_bigend(inputs=words, n_bytes=tx_data_len);
+
+            // Note: here, the validate process assumes an ECDSA signature, and r, s, v field
+            // Technically, the transaction type can determine the signature scheme.
+            let _is_legacy = is_legacy_tx(tx_data);
+            if (_is_legacy != FALSE) {
+                tempvar parity = (v - 2 * Constants.CHAIN_ID - 35);
```

Comment:
do this in `decode`, here the function should be version agnostic

Reply:
If we do it in decode, it's less clean because, either:
- decode takes as an argument `v` and returns `y_parity`, then when we call decode in __execute__ we have to provide a `v` argument which is out of the scope of `execute`